### ISSUE TITLE
Convert OTEL model to directly to storage model

### ISSUE THREAD

**Issue by pavolloffay** [jaegertracing/jaeger#2163]
Elasticsearch exporter converts OTEL span to Jaeger model and which is passed to `SpanWriter.WriteSpan`. The writer converts the span to ES storage model. We could consider converting the span directly to ES model to save some cycles.

This will require a separate convertor and will have to refactor ES span writer and expose a write method accepting Jaeger ES model.

**Comment by pavolloffay** [comment]
This has been done for Elasticsearch storage. 

It has to be done for:

- [ ] cassandra
- [ ] badger
- [ ] in-memory
- [ ] kafka

**Comment by yurishkuro** [comment]
We may want to consider if we want to support Cassandra with OTEL at all.

**Comment by yurishkuro** [comment]
@pavolloffay did you run any benchmark to see the impact of direct model xform?

**Comment by yurishkuro** [comment]
also, OTel trace model is slightly richer than Jaeger's, the translation to ES happened still to Jaeger's logical model? I think we should consider if we want to go to an extended, OTel-compatible model in the storage.

**Comment by pavolloffay** [comment]
I haven't run any perf tests. The new implementation basically skips one translation (OTLP-> Jaeger model) I haven't tested how much resources it takes.

> also, OTel trace model is slightly richer than Jaeger's, the translation to ES happened still to Jaeger's logical model?

The translation is done form OTLP to Jaeger ES data model directly. Agree we should have a look and extend storage models if necessary and also do appropriate handling in the UI.

**Comment by Vemmy124** [comment]
Hey! I have implemented unmarshalling OTLP from protobuf https://github.com/open-telemetry/opentelemetry-collector/pull/1948 (marshalling was already implemented), so it's now possible to use OTLP directly for in-memory and badger. Just need a new version of collector with my PR

**Comment by hdost** [comment]
If this is possible to go from Kafka (OTLP) -> Jaeger ES what configuration does that require on the kafka Side?

**Comment by pavolloffay** [comment]
yes at the moment it is possible to use OTEL collector with Kafka receiver with OTLP encoding and send data to Jaeger collector via Jaeger exporter. Jaeger collector will then store the data in ES.

**Comment by yurishkuro** [comment]
@pavolloffay i think it would be useful to add this to the APIs page in the docs, ie the API for the ingester which is not documented there.

**Comment by yurishkuro** [comment]
There is no useful information in this ticket, closing. It will come back in another form in #3500.
